### DISCUSSION TITLE
Project Management and Audit Trail Enhancement Proposal

### DISCUSSION THREAD

**Discussion by edjroz** [pokt-network/organization#83] [Proposals]
### Proposal Title 
Project Management and Audit Trail Enhancement Proposal
### Proposer
Luis: CEO
### Date(s)
09/02/2021
### Category
- [ ] Infrastructure
- [ ] Protocol/Blockchain Dev
- [ ] App Solutions
- [ ] Quality Assurance
- [X] Organization
### Contract
- [X] This process follows the Milestone Rank Vote Framework as discussed in the Pocket Network Corporation's `Organization Handbook`
- [ ] This process `DOES NOT` follow the Milestone Rank Vote Framework as discussed in the Pocket Network Corporation's `Organization Handbook`

If `DOES NOT` follow the Handbook's Milestone Rank Vote Framework, please describe in detail why below:
[Detailed Explanation]

### Detailed Summary of Proposal
Currently our workflow has 3 known issues.

1. Clear audit trail
2. Rank voting clarity
3. Development pace

In order to address point 1 And 2, this proposal focuses on using Github Discussions to store, proposals, voting, and results.

Github discussions offers clear threads of communication as well as allows for indexing of discussion through categories, we can add 1 or more categories to a discussion allowing for the ability to search.

The Github Discussion team is working on an API that could let us move discussions should it be necessary to migrate.

This targets specific pain points in the current workflow such as:

- Creation of issues, PRs,
- Direct push to the Repo to start any discussion.
- Eliminates the cumbersome method of creating a proposal/[date]/[name_of_proposal]
- Eliminates bloat of files on the repo.

Examples:
- #80 example Rank voting, close discussion with a clear decision marked.
- #79 example of proposal, open ended discussion, with negative feedback.
- #82 example of proposal, with template, and locked.

In order address point 3 we can use GitHub projects per member of the organization in conjunction with [this extension](https://github.com/banyan/github-story-points)

With minimal rewording of GitHub issues we are able to quantify development pace and keep a trail through the project of any member of the team. Each member can add tasks to their project signifying their bandwidth.

### Estimated Timeline
1 day to implement.

**Evidence and Justification for this timeline:**
Everything would exist within github, therefore we already have most of the needed stuff.

**Timeline Consultants**
- @oten91 
- @edjroz

**Comment by andrewnguyen22** [comment]
> is working on an API that could let us move discussions should it be necessary to migrate

This is the biggest thing for me... I'm interested to know more about the API

**Reply by edjroz** [reply]
Here is a [thread](https://github.com/github/feedback/discussions/43#discussioncomment-312134) for the official Discussions API in Discussion beta. (yes GH went a little meta using Discussions)

**Reply by edjroz** [reply]
Here's also the [docs](https://github.com/github/feedback/discussions/43#discussioncomment-339955)

**Reply by andrewnguyen22** [reply]
Looking through the docs i see nothing for exporting to a standard format.. Can you provide me this information?

**Reply by luyzdeleon** [reply]
The idea would be in case we need to switch, to build a script and extract it into markdown files @andrewnguyen22

**Reply by andrewnguyen22** [reply]
But how can you build a script to convert it to markdown? Where is the API to convert it to any format?

**Reply by andrewnguyen22** [reply]
If you can convert this conversation to markdown I'd be convinced XD

**Reply by luyzdeleon** [reply]
@andrewnguyen22 It's easy:

1. You create a single markdown file per discussion, every thread in the discussion is already in markdown.
2. You append the threads in markdown below.
3. You save the file with the discussion id mimicking the URL structure.

Voila. Migration in 3 easy steps.

**Comment by andrewnguyen22** [comment]
> Proposer Luis: CEO

Why did Eduardo submit this when Luis is the proposer?

**Reply by edjroz** [reply]
We had a long running meeting, Luis had to step out, and asked us to finish the proposal.
